File: 2022. Convert 1D Array Into 2D Array/2022. Convert 1D Array Into 2D Array.cpp
class Solution {
public:
    vector<vector<int>> construct2DArray(vector<int>& original, int m, int n) {
        if (m * n != original.size()) return {};
        
        vector<vector<int>> grid(m, vector<int>(n));
        for (int i = 0; i < m; ++i) {
            for (int j = 0; j < n; ++j) {
                // get linear index
                int linearIndex = i * n + j;
                grid[i][j] = original[linearIndex];
            }
        }
        
        return grid;
    }
};

File: 2022. Convert 1D Array Into 2D Array/2022. Convert 1D Array Into 2D Array.py
class Solution:
    def construct2DArray(self, original: List[int], m: int, n: int) -> List[List[int]]:
        if m * n != len(original):
            return []
        
        grid = [[0] * n for _ in range(m)]
        for i in range(m):
            for j in range(n):
                # get linear index
                linearIndex = i * n + j
                grid[i][j] = original[linearIndex]
        
        return grid